### PR TITLE
[kube-prometheus-stack] Bump dependencies

### PR DESCRIPTION
**Pull request by tun0**
#### What this PR does / why we need it:

Bump versions of dependencies. Primarily for https://github.com/prometheus-community/helm-charts/pull/1514.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by mrueg — DISMISSED** [review]

**Comment by QuentinBisson** [comment]
Thanks @monotek

**Review by monotek — COMMENTED** [review]
Aa this is a major version update in the dependency, is there anything we have to adjust here too?

**Comment by tun0** [comment]
> Aa this is a major version update in the dependency, is there anything we have to adjust here too?

I found the major version update a bit overkill (mostly some labeling stuff) and as such couldn't find any reason for other changes in this chart. Also, "works for me".

**Review by monotek — APPROVED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
Please fix liniting error.

Please also double check the upgrade path. In the prometheus chart we had to do a manual step, to be able to update kube-state-metrics, as there was an update to an immutable filed.

The kube-state-metrics deployment had to be deleted before the update to get it work. 
See: https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus#to-150

**Comment by tun0** [comment]
> Please fix liniting error.

Done (got called away while fixing that :grimacing:)

> Please also double check the upgrade path. In the prometheus chart we had to do a manual step, to be able to update kube-state-metrics, as there was an update to an immutable filed.
> 
> The kube-state-metrics deployment had to be deleted before the update to get it work. See: https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus#to-150

I might've fixed it on auto-pilot on my previous attempt, as indeed, a removal of the deployment (or in my case: statefulset) is indeed needed. I'll do a major version bump and add a bit of documentation to the README. Assuming that's the proper way to go?

File: charts/kube-prometheus-stack/README.md
Comment (monotek) [inline charts/kube-prometheus-stack/README.md:94]: I can't find an option in kube-state-metrics chart to run it as statefulset?

https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-state-metrics

So why do we need that other option?

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/README.md
Comment (tun0) [reply charts/kube-prometheus-stack/README.md:94]: See: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-state-metrics/templates/deployment.yaml#L1-L6

It depends on the `autosharding` setting.

File: charts/kube-prometheus-stack/README.md
Comment (monotek) [inline charts/kube-prometheus-stack/README.md:92]:
```suggestion
or if you use autosharding:
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/README.md
Comment (monotek) [reply charts/kube-prometheus-stack/README.md:94]: Sorry, web frontend commit messed up cto check. Will fix tomorrow or feel free to do it by yourself.

**Review by monotek — APPROVED** [review]

**Comment by tun0** [comment]
What else does it take to get this merged? The fixing of merge conflicts is getting annoying :wink:

**Comment by monotek** [comment]
As this is a new major version one of the chart maintainers should approve first.

**Review by monotek — APPROVED** [review]

**Pull request merged**
